atc/engine/check_delegate.go

```diff
@@ -95,7 +95,7 @@ func (d *checkDelegate) WaitToRun(ctx context.Context, scope db.ResourceConfigSc
 	}
 
 	var lock lock.Lock = lock.NoopLock{}
-	if d.plan.Resource != "" {
+	if d.plan.Resource != "" || d.plan.ResourceType != "" {
```

Comment:
now that we're checking for this combination in a few places, i wonder if adding a method like `(CheckPlan).IsPeriodic` would make it more readable?